plugin/pkg/admission/namespace/autoprovision/admission.go

```diff
@@ -83,6 +83,11 @@ func (p *Provision) Admit(a admission.Attributes) error {
 		Status: api.NamespaceStatus{},
 	}
 
+	// Don't actually create a namespace if the request is for a dry-run.
+	if a.IsDryRun() {
```

Comment:
This isn't the kind of implementation I expected. How about creating a filter for admission plugins (like we did for metrics) that we simply apply to every admission plugin we add to our chain.  Then the logic is contained to that filter and not distributed in each individual admission plugin?

Reply:
> This isn't the kind of implementation I expected. How about creating a filter for admission plugins (like we did for metrics) that we simply apply to every admission plugin we add to our chain. Then the logic is contained to that filter and not distributed in each individual admission plugin?

nm, that was a silly idea.